### PR TITLE
Fix #8634: Add error message if module is not specified and a file contains a external module indicator

### LINKED ISSUE #8634: --outFile is broken when module import is used
**TypeScript Version:** 
1.8.10

Node version :v4.2.2
OS: OSX

**Code**

**Main.ts**
```ts
/// <reference path="./typings/node/node.d.ts"/>

import * as http from 'http';

namespace Scratch
{
    export class ServerWrapper {

		public constructor() {
            this.server = http.createServer();
		}

		public Start(handler :(request :http.ClientRequest, response :http.ClientResponse)=>void) :void {
			this.server.listen(handler);
		}

		private server : http.Server;
	}
    var s = new Scratch.ServerWrapper();
    s.Start((req,res)=>{});
}
```
Compile with
```
tsc ./Main.ts --outFile /dev/stdout
```

**Expected behavior:** (what is emitted as Main.js when --outFile is not specified)
```js
"use strict";
var http = require('http');
var Scratch;
(function (Scratch) {
    var ServerWrapper = (function () {
        function ServerWrapper() {
            this.server = http.createServer();
        }
        ServerWrapper.prototype.Start = function (handler) {
            this.server.listen(handler);
        };
        return ServerWrapper;
    }());
    Scratch.ServerWrapper = ServerWrapper;
    var s = new Scratch.ServerWrapper();
    s.Start(function (req, res) { });
})(Scratch || (Scratch = {}));
```

**Actual behavior:** 
```
/// <reference path="./typings/node/node.d.ts"/>
```

### PR DESCRIPTION
**Pull request by mhegazy**
Fix #8634

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -447,7 +447,7 @@
         "category": "Error",
         "code": 1147
     },
-    "Cannot compile modules unless the '--module' flag is provided with a valid module type. Consider setting the 'module' compiler option in a 'tsconfig.json' file.": {
+    "Cannot use 'import', 'export' or module augmentation when compiling with '--module' is 'none'.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: imports, exports, or module augmentations

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -2751,7 +2751,12 @@
     "Resolving real path for '{0}', result '{1}'": {
         "category": "Message",
         "code": 6130
-    },    
+    },
+    "Cannot compile modules using option '{0}' unless the '--module' flag is provided with a valid module type.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: Be more explicit - we know exactly which module types are valid, so let's go with

`...unless the '--module' flag is 'amd' or 'system'.`

and maybe UMD if that's supported too - I always forget.

**Comment by mhegazy** [comment]
@DanielRosenwasser any more comments?

File: src/compiler/diagnosticMessages.json [hunk 3]
```diff
@@ -447,7 +447,7 @@
         "category": "Error",
         "code": 1147
     },
-    "Cannot compile modules unless the '--module' flag is provided with a valid module type. Consider setting the 'module' compiler option in a 'tsconfig.json' file.": {
+    "Cannot use imports, exports or module augmentations when '--module' is 'none'.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: Add a comma before "or"

**Comment by DanielRosenwasser** [comment]
Apart from the last correction, 👍

File: src/compiler/program.ts [hunk 4]
```diff
@@ -2089,8 +2089,14 @@ namespace ts {
             }
 
             // Cannot specify module gen that isn't amd or system with --out
-            if (outFile && options.module && !(options.module === ModuleKind.AMD || options.module === ModuleKind.System)) {
-                programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Only_amd_and_system_modules_are_supported_alongside_0, options.out ? "out" : "outFile"));
+            if (outFile) {
+                if (options.module && !(options.module === ModuleKind.AMD || options.module === ModuleKind.System)) {
+                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Only_amd_and_system_modules_are_supported_alongside_0, options.out ? "out" : "outFile"));
```
Comment (DanielRosenwasser) [inline src/compiler/program.ts]: I think you should give precedence to outFile in reporting the error if they're both specified. Also, do we error if both are specified?

Comment (mhegazy) [inline src/compiler/program.ts]: yes. there is another error for that.

**Comment by mhegazy** [comment]
merged manually. but @github does not seem to pick that up.

**Pull request closed without merging**